pages/fit_detail.py

```
@@ -0,0 +1,102 @@
+#     targets_df = get_targets()
+#     ship_fits = fitting_df[fitting_df['category_id'] == 6]
+#     merged_df = pd.merge(ship_fits, targets_df, on='ship_id', how='left')
+#     merged_df = merged_df.reset_index(drop=True)
+    
+#     return merged_df
+
+def main():
+    st.title("Fit Detail")
+    st.write("This is the fit detail page")
+
+    fitting_df = create_fit_df()
+    ship_fitting_df = fitting_df[fitting_df['category_id'] == 6]
+    ship_fitting_df = ship_fitting_df.sort_values(by='ship_name')
+    ship_fitting_df = ship_fitting_df.reset_index(drop=True)
+    available_fits = ship_fitting_df['ship_name'].unique().tolist()
+
+    # Item name filter - changed to selectbox for single selection
+    selected_ship = st.sidebar.selectbox(
+        "Select Ship",
+        options=[""] + available_fits,  # Add empty option to allow no selection
+        index=0,
+        format_func=lambda x: "All Fits" if x == "" else x
+    )
+
+    if selected_ship:
+        fit_ids = ship_fitting_df[ship_fitting_df['ship_name'] == selected_ship]
+        fit_ids = fit_ids['fit_id'].unique().tolist()
+ 
+        engine = get_local_mkt_engine()
+        fit_dict = {}
+        for fit_id in fit_ids:
+            with Session(engine) as session:
+                query = text("SELECT fit_name, fit_id FROM ship_targets WHERE fit_id = :fit_id")
+                result = session.execute(query, {'fit_id': fit_id})
+                df = pd.DataFrame(result.fetchall(), columns=result.keys())
+                fit_dict[df['fit_name'][0]] = [df['fit_id'][0]]
+        fit_options = [""] + list(fit_dict.keys())
+        if len(fit_ids) > 1:
+            selected_fit = st.sidebar.selectbox(
+                "Select Fit",
+                options=fit_options,
+                index=0,
+                format_func=lambda x: "All Fits" if x == "" else x
+            )
+        else:
+            selected_fit = fit_options[1]
+        
+        if selected_fit:
```

You unconditionally set selected_fit above, so need to check for it here I believe?

Unless fit_options[1] can be something invalid.

---

Good eye. I tested if selected ship before and we're still in the affirmative control flow of that conditional. But, we changed the value of options[1]. If something went wrong with the database call and it returned null values, this if selected fit would cause us to not proceed with creating the data frame. 

But, in that case it would probably be better to just raise let it raise an error here.